### PR TITLE
Integration test using master envoy and build infra improvements

### PR DESCRIPTION
**Pull request by kyessenov**
- expand the sample test driver to use split xDS and REST modes
- add the test driver to the CI to validate against the latest envoy
- add code coverage

### CODE REVIEW

**Comment by kyessenov** [comment]
Failure is due to dependency on #49 . Will fix depending which ones goes in first.

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,3 +10,8 @@ jobs:
       - run: make depend.install
       - run: make build
       - run: make test
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run: make integration.docker
+      - run: make cover
+      - run: curl -s https://codecov.io/bash | bash
```
Comment (rshriram) [inline .circleci/config.yml]: nit.. make sure this works.. I tried this in istio circleci and it had issues..

Comment (kyessenov) [reply .circleci/config.yml]: worked for me in the other PR.

**Comment by kyessenov** [comment]
@rshriram could you please take a look?

**Comment by kyessenov** [comment]
@junr03 can you please have a quick look?

**Pull request merged**